edge_api/src/main/java/com/example/edge_api/controller/InstrumentController.java

```diff
@@ -0,0 +1,44 @@
+package com.example.edge_api.controller;
+
+import com.example.edge_api.model.Instrument;
+import com.example.edge_api.model.Musician;
+import com.example.edge_api.model.Piece;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Arrays;
+import java.util.List;
+
+@RestController
+public class InstrumentController {
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @Value("${instrumentservice.baseurl}")
+    private String instrumentServiceBaseUrl;
+    @Value("${pieceservice.baseurl}")
+    private String pieceServiceBaseUrl;
+
+    @GetMapping("/instrument/{name}")
+    public Instrument getInstrumentWithMusiciansAndPieces(@PathVariable String name) {
```

Comment:
Does this not need to return a list? It just seems more logical that it would return a list of instruments that match the given string.

Reply:
Uitstellen tot front end